src/vivarium_gates_nutrition_optimization/components/disease.py

```diff
@@ -0,0 +1,412 @@
+from vivarium_public_health.disease import BaseDiseaseState
+from vivarium_public_health.utilities import is_non_zero
+
+
+class DiseaseState(BaseDiseaseState):
+    """State representing a disease in a state machine model."""
+
+    def __init__(
+        self,
+        cause: str,
+        get_data_functions: Dict[str, Callable] = None,
+        cleanup_function: Callable = None,
+        **kwargs,
+    ):
+        """
+        Parameters
+        ----------
+        cause : str
+            The name of this state.
+        disability_weight : pandas.DataFrame or float, optional
+            The amount of disability associated with this state.
+        prevalence_data : pandas.DataFrame, optional
+            The baseline occurrence of this state in a population.
+        dwell_time : pandas.DataFrame or pandas.Timedelta, optional
+            The minimum time a simulant exists in this state.
+        event_time_column : str, optional
+            The name of a column to track the last time this state was entered.
+        event_count_column : str, optional
+            The name of a column to track the number of times this state was entered.
+        side_effect_function : callable, optional
+            A function to be called when this state is entered.
+        """
+        super().__init__(cause, **kwargs)
+
+        self.excess_mortality_rate_pipeline_name = f"{self.state_id}.excess_mortality_rate"
+        self.excess_mortality_rate_paf_pipeline_name = (
+            f"{self.excess_mortality_rate_pipeline_name}.paf"
+        )
+
+        self._get_data_functions = (
+            get_data_functions if get_data_functions is not None else {}
+        )
+        self.cleanup_function = cleanup_function
+
+        if self.cause is None and not set(self._get_data_functions.keys()).issuperset(
+            ["disability_weight", "dwell_time", "prevalence"]
+        ):
+            raise ValueError(
+                "If you do not provide a cause, you must supply"
+                "custom data gathering functions for disability_weight, prevalence, and dwell_time."
+            )
+
+    #################
+    # Setup methods #
+    #################
+
+    # noinspection PyAttributeOutsideInit
+    def setup(self, builder: Builder) -> None:
+        """Performs this component's simulation setup.
+
+        Parameters
+        ----------
+        builder : `engine.Builder`
+            Interface to several simulation tools.
+        """
+        super().setup(builder)
+
+        self.prevalence = self.get_prevalence_table(builder)
+
+        self.birth_prevalence = self.get_birth_prevalence_table(builder)
+
+        self.dwell_time = self.get_dwell_time_pipeline(builder)
+
+        self.has_disability, self.base_disability_weight = self.get_disability_weight_table(
+            builder
+        )
+
+        self.disability_weight = self.get_disability_weight_pipeline(builder)
```

Comment:
Nit: I don't think we need a separate `get_disability_weight_table` function. `get_disability_weight_pipeline` should be sufficient.